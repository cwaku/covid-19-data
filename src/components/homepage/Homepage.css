body {
  background-image: url(../../images/covidBG.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  background-attachment: fixed;
}