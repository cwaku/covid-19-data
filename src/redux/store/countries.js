import moment from 'moment';
import { createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
// set axios default url
axios.defaults.baseURL = 'https://api.covid19tracking.narrativa.com/api/';

const date = moment(new Date()).format('YYYY-MM-DD');

const countriesSlice = createSlice({
  name: 'countries',
  initialState: {
    countries: [],
    selectedCountry: '',
    searchCountry: '',
  },
  reducers: {
    getCountriesSuccess: (state, action) => ({ ...state, countries: action.payload }),
    getSelectedCountrySuccess: (state, action) => ({ ...state, selectedCountry: action.payload }),
    getSearchedCountrySuccess: (state, action) => ({ ...state, searchCountry: action.payload }),
    getCountriesFailure: (state, action) => ({ ...state, countries: action.payload }),
  },
});

export default countriesSlice.reducer;

// Actions

export const {
  getCountriesSuccess,
  getSelectedCountrySuccess,
  getSearchedCountrySuccess,
  getCountriesFailure,
} = countriesSlice.actions;

export const fetchCountries = () => async (dispatch) => {
  try {
    const response = await axios.get(date);
    if (response) {
      const { dates, total } = response.data;
      const { countries } = dates[date];
      const countriesList = Object.values(countries);

      const singleCountry = countriesList.map((country) => (
        {
          name: country.name,
          confirmedCases: country.today_confirmed,
          deaths: country.today_deaths,
          recovered: country.today_recovered,
          active: country.today_open_cases,
          id: country.id,
        }
      ));

      const totalConfirmedCases = total.today_confirmed;
      const allData = { singleCountry, totalConfirmedCases };
      dispatch(getCountriesSuccess(allData));
    }
  } catch (error) {
    dispatch(getCountriesFailure(error));
  }
};
