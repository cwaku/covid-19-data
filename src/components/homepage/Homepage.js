import Container from '@mui/material/Container';
import Countries from './Country';
import './Homepage.css';

const Homepage = () => {
  return (
    <Container>
      <h1>Homepage</h1>
      <Countries />
    </Container>
  );
};

export default Homepage;