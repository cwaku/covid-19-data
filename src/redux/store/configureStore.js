import { configureStore } from '@reduxjs/toolkit'
import thunk from 'redux-thunk'
import logger from 'redux-logger'
import { combineReducers } from 'redux'
import countries from './countries'

const reducer = combineReducers({
  // reducers go here
  countries,
})

const store = configureStore({
  reducer,
  middleware: [thunk, logger],
})

export default store;