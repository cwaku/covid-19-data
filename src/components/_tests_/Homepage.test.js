import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../../redux/store/configureStore';
import Homepage from '../homepage/Homepage';

describe('Test Homepage component', () => {
  it('should render the Homepage component correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={store}>
          <Homepage />
        </Provider>
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
