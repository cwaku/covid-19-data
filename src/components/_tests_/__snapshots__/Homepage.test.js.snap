// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Homepage component should render the Homepage component correctly 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg main-container css-1oqqzyl-MuiContainer-root"
  >
    <nav
      class="navbar"
    >
      <div
        class="navbar-container"
      >
        <div
          class="navbar-logo"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium navbar-icon css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowBackIosOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17.51 3.87 15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
            />
          </svg>
        </div>
        <h3>
          Global cases
        </h3>
        <div
          class="navbar-menu"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MicOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 14c1.66 0 3-1.34 3-3V5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3z"
            />
            <path
              d="M17 11c0 2.76-2.24 5-5 5s-5-2.24-5-5H5c0 3.53 2.61 6.43 6 6.92V21h2v-3.08c3.39-.49 6-3.39 6-6.92h-2z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SettingsOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.09-.16-.26-.25-.44-.25-.06 0-.12.01-.17.03l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.06-.02-.12-.03-.18-.03-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.09.16.26.25.44.25.06 0 .12-.01.17-.03l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7.14 1.13zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
            />
          </svg>
        </div>
      </div>
    </nav>
    <div
      class="banner MuiBox-root css-8atqhb"
    >
      <h1>
        COVID-19
      </h1>
      <h2>
        Coronavirus disease (COVID-19)
      </h2>
      <h2
        class="cases"
      >
        Total confirmed cases:
      </h2>
      <div
        class="banner-text"
      >
        <h3
          class="no-margin"
        >
          Coronavirus disease (COVID-19) is an infectious disease caused by a newly discovered coronavirus.
        </h3>
        <h3
          class="no-margin"
        >
          Most people infected with the COVID-19 virus will experience mild to moderate respiratory illness and recover without requiring special treatment.
        </h3>
        <h3
          class="no-margin"
        >
          The COVID-19 virus spreads primarily through droplets of saliva or discharge from the nose when an infected person coughs or sneezes, so itâ€™s important that you also practice respiratory etiquette (for example, by coughing into a flexed elbow).
        </h3>
        <h3
          class="no-margin"
        >
          At this time, there are no specific vaccines or treatments for COVID-19.
        </h3>
        <h3
          class="no-margin"
        >
          However, there are many ongoing clinical trials evaluating potential treatments. WHO will continue to provide updated information as soon as clinical findings become available.
        </h3>
        <h3
          class="no-margin"
        >
          The World Health Organization (WHO) is the authority on the COVID-19 pandemic.
        </h3>
        <h3
          class="no-margin"
        >
          Read more about the COVID-19 pandemic at
          <a
            href="https://www.who.int/emergencies/diseases/novel-coronavirus-2019"
          >
            https://www.who.int/emergencies/diseases/novel-coronavirus-2019
          </a>
        </h3>
      </div>
    </div>
    <div
      class="MuiBox-root css-8atqhb"
    >
      <h2
        class="stats"
      >
        STATS BY COUNTRY:
      </h2>
      <input
        class="search-input"
        placeholder="Search for a country"
        type="text"
        value=""
      />
      <div
        class="MuiGrid-root MuiGrid-container css-10mzr36-MuiGrid-root"
      />
    </div>
  </div>
</div>
`;
